service/project_service/upload.go

```diff
@@ -0,0 +1,64 @@
+package service
+
+import (
+	"encoding/base64"
+	"io"
+	"unicode/utf8"
+
+	"github.com/go-errors/errors"
+	"github.com/guregu/null"
+
+	"github.com/torinos-io/api/type/model"
+)
+
+// UploadRequest holds uploaded files
+type UploadRequest struct {
+	CartfileContent    io.Reader `json:"-"`
+	PodfileLockContent io.Reader `json:"-"`
+	RepositoryName     string    `json:"repository_name"`
+}
+
+// Upload uploads files to Analyze service
+func (s *service) Upload(userID null.Int, req *UploadRequest) (*model.Project, error) {
+	if req.CartfileContent == nil && req.PodfileLockContent == nil {
+		return nil, errors.New("CartFile and PodFile are both empty")
+	}
+
+	data := &model.UploadedData{}
+
+	if s, err := toBase64(req.PodfileLockContent); utf8.RuneCountInString(s) != 0 && err == nil {
```

Comment:
> `utf8.RuneCountInString(s) != 0`

You don't need to use RuneCountInString generally, use `len()`.

Btw, this can be written `s != ""`